### PR TITLE
Fixes #24125 - Show error when libvirt not reachable

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: app/models/compute_resources/foreman/model/libvirt.rb [hunk 1]
```diff
@@ -230,6 +230,12 @@ def normalize_vm_attrs(vm_attrs)
 
     protected
 
+    def libvirt_connection_error
+      msg = _('Unable to connect to libvirt due to: %s. ') % errors.full_messages.join(', ')
+      msg += _('Please make sure your libvirt compute resource is reachable and that you have appropriate access permissions.')
+      raise Foreman::Exception.new(msg)
```
Comment (lzap) [inline app/models/compute_resources/foreman/model/libvirt.rb]: The interpolation must go into exception as a parameter. Also use just one string:

```ruby
msg = N_("Full message")
raise Foreman::Exception.new(msg, errors.full_messages.join(', '))
```

**Review by lzap — DISMISSED** [review]
Lots of broken tests, I think they are relevant.

**Comment by xprazak2** [comment]
Updated and fixed tests.

**Review by dLobatog — CHANGES_REQUESTED** [review]
Thanks, this works pretty well. I see that long error messages are clipped on the UI though (I know this is for the UI but still it'll be visible in the UI). Can you fix that? Should be ready to merge after that.

You may reproduce this easily by raising an exception with a long text on `app/models/compute_resources/foreman/model/libvirt.rb#test_connection`, for example: `raise StandardError.new("ohnoes. this libvirt is broken")` 

screenshot from 2018-07-05 16-33-32


Thank you @xprazak2

**Comment by xprazak2** [comment]
Updated, truncating messages for alerts makes not sense since there is no tooltip.

**Comment by ares** [comment]
tests are still failing, @xprazak2 please take a look, thanks

```
--- expected
+++ actual
@@ -1,2 +1,2 @@
-"<div class=\"alert alert-danger \"><span class=\"pficon pficon-error-circle-o \"></span> <span class=\"text\"><span data-original-title=\"Failure: ActiveRecord::RecordNotFound\" rel=\"twipsy\">Failure: ActiveRecord::RecordNotFound</span></span></div>
+"<div class=\"alert alert-danger \"><span class=\"pficon pficon-error-circle-o \"></span> <span class=\"text\"><span>Failure: ActiveRecord::RecordNotFound</span></span></div>
 "
 (Minitest::Assertion)
/var/lib/workspace/workspace/test_develop_pr_core@3/database/postgresql/ruby/2.4/slave/fast/test/controllers/hosts_controller_test.rb:257    
```

**Comment by xprazak2** [comment]
[test katello]

**Comment by xprazak2** [comment]
[test katello]

Docker Meta Tags test is known to fail sometimes and this probably does not have effect on activation key audits either.

**Review by dLobatog — APPROVED** [review]

File: app/models/compute_resources/foreman/model/libvirt.rb [hunk 2]
```diff
@@ -230,6 +230,11 @@ def normalize_vm_attrs(vm_attrs)
 
     protected
 
+    def libvirt_connection_error
+      msg = N_('Unable to connect to libvirt due to: %s. Please make sure your libvirt compute resource is reachable and that you have appropriate access permissions.')
+      raise Foreman::Exception.new(msg, errors.full_messages.join(', '))
```
Comment (tbrisker) [inline app/models/compute_resources/foreman/model/libvirt.rb:235]: I'm not sure how this passed PR tests, but it is now failing on jenkins - https://ci.theforeman.org/job/test_develop/3603/database=postgresql,ruby=2.3,slave=fast/testReport/junit/(root)/HostJSTest__create%20new%20host%20page/test_0003_choosing_a_hostgroup_with_compute_resource_works/

**Pull request merged**